src/features/courses/course-route.ts

```diff
@@ -438,15 +483,24 @@ router.get('/question/:id',
 
         const params = req.params as GetQuestionRequest.params;
         try {
-            // TODO handle not found case
-            const question = await courseController.getQuestion({
-                questionId: params.id,
-                userId: session.userId,
-                formURL: req.originalUrl,
-                role: user.roleId,
-                readonly: req.query.readonly,
-                workbookId: req.query.workbookId,
-            });
+            let question;
+            if (req.query.userId) {
+                question = await courseController.getQuestionWithoutRenderer({
+                    id: params.id,
+                    userId: req.query.userId,
+                });
+            } else {
+                // TODO handle not found case
+                question = await courseController.getQuestion({
+                    questionId: params.id,
+                    // TODO: Is there a better way to handle the authentication for this?
```

Comment:
What do you mean? I think grabbing from the session is probably the best (and most secure way)
You don't have to check as many permissions if it comes from the session (i.e. if we allowed another user to fetch a question as if they were the student like a professor seeing what they see)

Reply:
That was actually a slightly out-of-date comment. It was meant to be about using `req.query.userId` in the conditional above this.